src/Conversion/ONNXToTorch/ONNXToTorchCommon.cpp

```diff
@@ -16,34 +16,80 @@
         .getResult(0);
+  };
+  addSourceMaterialization(addUnrealizedCast);
+  addTargetMaterialization(addUnrealizedCast);
+
+  /// String conversion
+  addConversion([](StringType stringType) {
+    return Torch::StringType::get(stringType.getContext());
   });
 
-  addTargetMaterialization([&](OpBuilder &builder, Type resultType,
-                               ValueRange inputs,
-                               Location loc) -> Optional<Value> {
-    if (inputs.size() != 1)
-      return llvm::None;
-    return builder.create<UnrealizedConversionCastOp>(loc, resultType, inputs)
-        .getResult(0);
+  /// Tensor conversion
+  addConversion([](TensorType type) -> Optional<Type> {
+    return Torch::ValueTensorType::get(
+      type.getContext(), type.getShape(), type.getElementType());
+  });
+  auto addTensorCast = [](OpBuilder &builder, TensorType type,
+                          ValueRange inputs, Location loc) -> Value {
+    assert(inputs.size() == 1);
+    assert(inputs[0].getType().isa<Torch::BaseTensorType>());
+    return builder.create<torch::TorchConversion::ToBuiltinTensorOp>(loc, inputs[0]);
+  };
+  addSourceMaterialization(addTensorCast);
+  addArgumentMaterialization(addTensorCast);
+  addTargetMaterialization([](OpBuilder &builder, Torch::ValueTensorType type,
+                              ValueRange inputs, Location loc) -> Value {
+    assert(inputs.size() == 1);
+    assert(inputs[0].getType().isa<TensorType>());
+    return builder.create<torch::TorchConversion::FromBuiltinTensorOp>(
+      loc, type, inputs[0]);
+  });
+
+  /// Create tensor to value tensor type conversion and ensure that
+  /// we always use signed integer types. i64 -> si64 conversion.
+  addConversion([](RankedTensorType type) -> Type {
+    Type elementType = type.cast<TensorType>().getElementType();
+    if (type.getElementType().isSignlessInteger(64)) {
+      elementType = IntegerType::get(type.getContext(), 64, IntegerType::Signed);
+    }
+    return Torch::ValueTensorType::get(type.getContext(),
+      type.getShape(), elementType);
+  });
+  addTargetMaterialization([](OpBuilder &builder,
+                              RankedTensorType type, ValueRange inputs,
+                              Location loc) -> Optional<Value> {
+    /// Other builtin integer types could be handled by other materializers.
+    assert(inputs.size() == 1);
+    assert(inputs[0].getType().isa<RankedTensorType>());
+    Type elementType = type.cast<TensorType>().getElementType();
+    if (type.getElementType().isSignlessInteger(64)) {
+      elementType = IntegerType::get(type.getContext(), 64, IntegerType::Unsigned);
+    }
+    return builder.create<UnrealizedConversionCastOp>(loc,
+      elementType, inputs).getResult(0);
   });
+  auto addValueTensorCast = [](OpBuilder &builder, Torch::ValueTensorType type,
+                               ValueRange inputs, Location loc) -> Optional<Value> {
+    assert(inputs.size() == 1);
```

Comment:
Since we don't have conversions that return multiple types, this materialization should never be called with more than 1 inputs. However, if we add more type conversion, it may be better to turn the assert into a RTC that returns None.